src/udev/device.rs

```diff
@@ -653,10 +653,21 @@ pub struct Device {
 }
 
 impl Device {
+    /// Returns a path to the sysfs device link, such as
+    /// "/sys/devices/pci0000:00/0000:00:08.1/0000:09:00.3/usb1/1-3/1-3:1.2/0003:0B05:1ABE.0003/hidraw/hidraw2/device"
+    fn sysnode_device_link(&self) -> String {
+        let s = self.path.as_str();
+        match s.ends_with("/device") {
```

Comment:
Is this actually the correct thing to do? In changes below you change from this:
```rust
        let path = format!("/sys{}/device/id/vendor", self.path.clone());
```
to this:
```rust
        let path = format!("{}/id/vendor", self.sysnode_device_link());
```
The path now has `device` appended to it since it didn't end with it. Only `get_parent_device_name()` expected this.

I would suggest adding some unit tests. Or drop this change completely since the commit apparently addresses only the `u64 change and not this.

In fact please drop all the changes that aren't `u64`. You've changed the behaviour unnecessarily

Reply:
Agree, only the bugfix is needed here.